### ISSUE TITLE
Each interactive element's role should match its function 

### ISSUE THREAD

**Issue by blnkt** [ManifoldScholar/manifold#1079]
All interactive html elements should have an appropriate role, applied either through the implicit aria semantics of said element, or explicitly through the role attribute.  Additionally this role should never conflict with the implicit aria semantics of said element.  Furthermore elements should be correctly nested according to their roles (i.e. buttons should not be nested inside of buttons, links should not be nested inside of links, etc.)

See https://www.w3.org/TR/html-aria/#docconformance
